lib/parser/provenance.cc

```diff
@@ -328,18 +401,43 @@ std::optional<ProvenanceRange> CookedSource::GetProvenanceRange(
   return {ProvenanceRange{first.start(), last.start() - first.start()}};
 }
 
+std::optional<CharBlock> CookedSource::GetCharBlock(
+    ProvenanceRange range) const {
+  if (auto to{invertedMap_.Map(range)}) {
```

Comment:
Should this assert that `invertedMap_` has been built?

Reply:
Maybe.  Or maybe it should be non-`const` and build the inverted map on the first reference.